x-pack/plugins/files/public/components/file_picker/file_picker_state.ts

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  map,
+  tap,
+  from,
+  switchMap,
+  Observable,
+  shareReplay,
+  debounceTime,
+  Subscription,
+  combineLatest,
+  BehaviorSubject,
+  distinctUntilChanged,
+} from 'rxjs';
+import { FileJSON } from '../../../common';
+import { FilesClient } from '../../types';
+
+function naivelyFuzzify(query: string): string {
+  return query.includes('*') ? query : `*${query}*`;
+}
+
+export class FilePickerState {
```

Comment:
In the future we can think about adding caching layer to files (and saved objects in general, as part of content management); so that when you open the file picker modal for the second time (or, when you delete the search query), the file list appears immediately.

Reply:
Yes, I was thinking this too. I think we should be able to do this quite easily with the `QueryClient` by `@tanstack/react-query`. Although rolling our own (if we want to) should be straightforward too.